chk/modules/http/presentation.py

```diff
@@ -3,23 +3,36 @@
 """
 
 
-def make_displayable(response: dict[str, object]) -> str:
-    def headers(res) -> str:
-        """Headers"""
-        return '{}'.format(
-            '\r\n'.join('{}: {}'.format(k, v) for k, v in res.get('headers').items()),
-        )
+class Presentation:
+    """Handle presentation of the outputs of chkware commands."""
 
-    if response.get('have_all'):
-        summary = '{} {} {}'.format(
-            response.get('version'),
-            response.get('code'),
-            response.get('reason'),
-        )
+    @staticmethod
+    def displayable_result(response: dict[str, object]) -> str:
+        """Return result in presentable format."""
+        prefix = '[RESULT]'
 
-        return '{}\r\n{}\r\n\r\n{}'.format(summary, headers(response), response.get('body'))
-    else:
-        response.pop('have_all')
-        for _, val in response.items():
-            if val:
-                return str(val)
+        def headers(res) -> str:
+            """Headers"""
+            return '{}'.format(
+                '\r\n'.join('{}: {}'.format(k, v) for k, v in res.get('headers').items()),
+            )
+
+        if response.get('have_all'):
+            summary = '{} {} {}'.format(
+                response.get('version'),
+                response.get('code'),
+                response.get('reason'),
+            )
+
+            return '{}\r\n{}\r\n{}\r\n\r\n{}'.format(prefix, summary, headers(response), response.get('body'))
+        else:
+            response.pop('have_all')
+            for _, val in response.items():
+                if val:
+                    return '{}\r\n{}'.format(prefix, str(val))
+
+    @staticmethod
+    def displayable_execution_summary(filepath: str) -> str:
+        """Return execution summary in presentable format."""
+        summary = 'File: {}\r\n\nExecuting request\r\n\n- Making request [Success]\r\n===='
+        return summary.format(filepath)
```

Comment:
What do you show when the request if failed? due to network fail, or ...

Reply:
right, I was also thinking about that. Can you please point out where I can get the status if request succeeded or not? As of now, I printed the summary after getting the response. In this case, we can say the request is a success otherwise it would have exited before. Similarly for `displayable_result`

Reply:
error response could be [like this](https://github.com/chkware/cli/issues/114#issuecomment-1193073284)

> Can you please point out where I can get the status if request succeeded or not?

What's your idea?

Reply:
if I think from a higher level, `handle_worker(http_spec)` will throw an exception if there is any problem while making the request. We can capture the exception using the `try...except` block and take the decision to show an appropriate summary.
```python
    try:
        response = handle_worker(http_spec)
        display_res(response)
    except as ex:
        display_res(ex)
```

However, it might be better if we can capture the exception exactly when it was raised.